### ISSUE TITLE
Can't break pcap_loop() with pcap_breakloop() if there is no traffic

### ISSUE THREAD

**Issue by milabs** [the-tcpdump-group/libpcap#903]
I have a problem with `libpcap-1.9.1` on Linux (kernel 5.4.15).

It seems that `poll()` doesn't returns if there is no traffic being captured. I've seen that there is `-1` used as a timeout value for the poll (despite the timeout value set with `pcap_set_timeout`) which seems **strange** for me but I'm not familiar with libpcap code base so let it be there...

My current code is like:
```
p = pcap_create(...)
pcap_set_snaplen(p, ...)
pcap_set_timeout(p, 1000)
pcap_ativate(p)
pcap_compile(p, ..., "ip host 8.8.8.8", 0, PCAP_NETMASK_UNKNOWN)
pcap_set_filter(p, ...)
pcap_loop(p, -1, ...)
```

Then, there are two test cases I have:
1) Calling `pcap_brakloop()` after a while in parallel with `ping -s64000 -f 8.8.8.8` (so, the traffic is there). The loop breaks, test **PASSED**.
2) Calling `pcap_brakloop()` after a while (so, there is no traffic there). The loop doesn't break, test **FAILED**.

Adding `pcap_set_immediate_mode(p, 1)` helps to solve the problem, but I don't want it to be there because using immediate mode is to CPU consuming for me.

Any ideas?

**Comment by guyharris** [comment]
Try the current master branch of libpcap, in which:

- a Linux interface `pcap_t` has both a PF_PACKET socket on which to capture *and* an "event" file descriptor;
- the `poll()` call that waits for new packets to arrive waits on *both* of those descriptors;
- `pcap_breakloop()` posts an event on the "event" file descriptor, so that it wakes up the `poll()`.

**Comment by milabs** [comment]
@guyharris Hey, it works. Could you tell me how stable is the master vs 1.9.1 or what was the commit which fixes the behavior I had so I will cherry-pick it over the 1.9.1?

**Comment by milabs** [comment]
Ok, it seems there is the original commit fixing that issue 5c8b13d3e87542527ed9a3a79fb0f9b2edb74df1. August 2018! I wonder why it's not in the 1.9.1 ...

**Comment by infrastation** [comment]
That fix was included in libpcap 1.10.0, hence the issue is resolved, closing.
